### PR TITLE
Add VSCode and EditorConfig files

### PR DESCRIPTION
**Pull request by ob-stripe**
r? @rattrayalex-stripe

### CODE REVIEW

**Comment by rattrayalex-stripe** [comment]
I thought I added this in my branch? they are slightly different, and should probably be merged...

File: .vscode/extensions.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+    "recommendations": [
+        // JS Language plugins (~required)
+        "dbaeumer.vscode-eslint",
+        "joshpeng.sublime-babel-vscode",
+        "esbenp.prettier-vscode",
+
+        // Highly recommended
+        "eamodio.gitlens",
+        "ziyasal.vscode-open-in-github",
```
Comment (rattrayalex-stripe) [inline .vscode/extensions.json]: let's remove these two, so that contributors don't get cluttered.

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+  // Show the path in the top window bar.
+  "window.title": "${rootName}${separator}${activeEditorMedium}",
+
+  // Formatting
+  "editor.formatOnSave": true,
```
Comment (rattrayalex-stripe) [inline .vscode/settings.json]: there is some beautify excludes that we have in others like prettier-poet, can you copy those over here?

Comment (ob-stripe) [reply .vscode/settings.json]: Hm, I can't find any `beautify.ignore` setting in VSCode with my current configuration. I assume it's a setting for a different extension is not in the list of recommended extensions, but that, if installed separately, would interfere with ESLint auto-formatting? Is that correct?

Comment (rattrayalex-stripe) [reply .vscode/settings.json]: yeah, with the prettier auto-formatting. 

We want this:
```
"beautify.ignore": ["**/*.js", "**/*.md"],
```

Comment (rattrayalex-stripe) [reply .vscode/settings.json]: Eh, I'll just add that to your PR.

**Pull request merged**